cmd/e2etest/chain/bsc/api.go

```diff
@@ -127,9 +127,7 @@ func (a *api) GetCoinBalance(coinName string, addr string) (*chain.CoinBalance,
 	if !strings.Contains(addr, "btp://") {
 		return nil, errors.New("Address should be BTP address. Use GetBTPAddress(hexAddr)")
 	}
-	if !strings.Contains(addr, ".bsc") {
```

Comment:
The conditional check ensures that the address whose balance is being checked is a BTP address of the specific chain. For example, here ".bsc" is checked to ensure that the address is of the format btp://0xnid.bsc/hex_address

Avoiding this conditional would mean, this bsc.GetCoinBalance can be called for recipient address of the form: btp://0x2.icon/hx123, which does not make sense because you can not get balance of an ICON address by querying the bsc chain.

As for the coinName, it is processed as it is without any conditional checks. So even if token names do not adhere to the format btp-<nid>-<coinSymbol>, this GetCoinBalance() method will still work.

This applies to all other changes made in this commit.

Reply:
New coin/token name format doesn't have this part. This validation always fails. 
Old name format: btp-0x38.bsc-BNB
New name format: BNB

Reply:
```
func (a *api) GetCoinBalance(coinName string, addr string) (*chain.CoinBalance, error) {
	if !strings.Contains(addr, "btp://") {
		return nil, errors.New("Address should be BTP address. Use GetBTPAddress(hexAddr)")
	}
	if !strings.Contains(addr, ".bsc") {
		return nil, fmt.Errorf("Address should be BTP address of account in native chain bsc. Got %v", addr)
	}
	splts := strings.Split(addr, "/")
	address := splts[len(splts)-1]
	return a.requester.getCoinBalance(coinName, address)
}
```

Arg: coinName = "BNB"
a.GetCoinBalance("BNB", some_btp_address) => a.requester.getCoinBalance("BNB", some_hex_address)

There is no validation checks for the field coinName inside the function. If so, how will it fail ?
The line that you have omitted out is validation check for addr not coinName. 
Not sure if I'm not getting something here.

Reply:
Hi @hemz10 -- could you also review this PR ? Is the suggested change on GetCoinBalance() necessary for the function to be able to handle coinNames of the type "BNB" ?

Reply:
Hi @manishbista28 I agree with you. Validation in the function is only for addr and not for coinName.

Reply:
I got it, I was misled by the meaning of GodAddr